### PR TITLE
Abandon UnstableClient idea in favor of symbol references

### LINKED ISSUE #718: Move unstable methods back into Client/ObjectSet
We want them hidden behind a symbol to reference the function so they are hard to consume but removes the casting pain we have seen internally.

### PR DESCRIPTION
**Pull request by ericanderson**
Fixes #718

### CODE REVIEW

File: packages/client/src/objectSet/createObjectSet.ts [hunk 1]
```diff
@@ -227,6 +230,19 @@ export function createObjectSet<Q extends ObjectOrInterfaceDefinition>(
         ) as Result<Osdk<Q>>;
       }
       : undefined) as ObjectSet<Q>["fetchOneWithErrors"],
+
+    [__EXPERIMENTAL__NOT_SUPPORTED_YET_subscribe]: (
```
Comment (ericanderson) [inline packages/client/src/objectSet/createObjectSet.ts:234]: Example usage

File: packages/e2e.sandbox.catchall/src/runSubscriptionsTest.ts [hunk 2]
```diff
@@ -38,9 +41,13 @@ export function runSubscriptionsTest() {
 
   client(Employee).where({
```
Comment (ericanderson) [inline packages/e2e.sandbox.catchall/src/runSubscriptionsTest.ts:42]: Example usage

File: packages/client/src/Client.ts [hunk 3]
```diff
@@ -40,14 +49,19 @@ export type CheckVersionBound<Q> = Q extends VersionBound<infer V> ? (
...
       }
-    // Q
   )
   : Q;
 
 export interface Client extends SharedClient<MinimalClient> {
   <Q extends MinObjectDef<any, any>>(
     o: Q,
-  ): CompileTimeMetadata<Q>["objectSet"];
+  ): unknown extends CompileTimeMetadata<Q>["objectSet"] ? ObjectSet<Q>
```
Comment (ssanjay1) [inline packages/client/src/Client.ts:58]: Why did we need this change?

File: packages/client/src/Client.ts [hunk 4]
```diff
@@ -71,6 +85,34 @@ export interface Client extends SharedClient<MinimalClient> {
...
   >;
+
+  /**
+   * WARNING. THIS METHOD IS EXPERIMENTAL AND NOT SUPPORTED YET.
+   *
+   * It may change at any time and does not follow semantic versioning. Use at your own risk.
+   *
+   *  @alpha
+   */
+  readonly [__EXPERIMENTAL__NOT_SUPPORTED_YET__preexistingObjectSet]: <
```
Comment (ssanjay1) [inline packages/client/src/Client.ts:96]: This pre-existing objectset thing is new right? What use case are we adding this for?

File: etc/client.api.report.api.md [hunk 5]
```diff
@@ -475,6 +475,10 @@ export type ObjectMetadata = {
 
 // @public (undocumented)
 export interface ObjectSet<Q extends ObjectOrInterfaceDefinition = any, _UNUSED = any> extends MinimalObjectSet<Q> {
+    // Warning: (ae-forgotten-export) The symbol "EXPERIMENTAL_ObjectSetListener" needs to be exported by the entry point index.d.ts
+    //
+    // @alpha
+    readonly [__EXPERIMENTAL__NOT_SUPPORTED_YET_subscribe]: (listener: EXPERIMENTAL_ObjectSetListener<Q>) => () => unknown;
```
Comment (ssanjay1) [inline etc/client.api.report.api.md:481]: Any reason you didn't just want to call these all `__EXPERIMENTAL_subscribe` etc.?

**Review by ssanjay1 — APPROVED** [review]
Approved to unblock, with questions

File: packages/client/src/Client.ts (hunk 3 above)
Comment (ericanderson) [reply packages/client/src/Client.ts:58]: I think this is similar to the issue you saw recently where the type complexity ends up failing us. If I recall it was no longer able to determine if the primaryKey was the same type

File: packages/client/src/Client.ts (hunk 4 above)
Comment (ericanderson) [reply packages/client/src/Client.ts:96]: Its not new, we had it before. Its for people who have a rid for an object set they want to start with.

File: etc/client.api.report.api.md (hunk 5 above)
Comment (ericanderson) [reply etc/client.api.report.api.md:481]: scarier the better! Also we can rename it as much as we want so if you like it smaller we can do that later.

File: packages/client/src/Client.ts (hunk 4 above)
Comment (ssanjay1) [reply packages/client/src/Client.ts:96]: Oh ok sounds good then

File: packages/client/src/Client.ts (hunk 3 above)
Comment (ssanjay1) [reply packages/client/src/Client.ts:58]: Ah yea that was the same issue I ran into

**Pull request merged**